handlers/handler-flowcontrol-qps/src/main/java/org/apache/servicecomb/qps/QpsControllerManager.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.qps;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import org.apache.servicecomb.foundation.common.concurrent.ConcurrentHashMapEx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.config.DynamicProperty;
+
+public class QpsControllerManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(QpsControllerManager.class);
+
+  private static final Pattern QUALIFIED_KEY_CHECKER = Pattern.compile("^[^.]+\\.[^.]+\\.[^.]+$");
+
+  /**
+   * Describe the relationship between configuration and qpsController.
+   */
+  protected final Map<String, QpsController> configQpsControllerMap = new ConcurrentHashMapEx<>();
+
+  /**
+   * Describe the relationship between qualifiedKey(format is "microservice.schema.operation") and qpsController.
+   */
+  protected final Map<String, QpsController> qualifiedNameControllerMap = new ConcurrentHashMapEx<>();
+
+  protected QpsController globalQpsController;
+
+  public static final String SEPARATOR = ".";
+
+  private String configKeyPrefix;
+
+  public QpsController getOrCreate(String key) {
+    return qualifiedNameControllerMap.computeIfAbsent(key, this::create);
+  }
+
+  /**
+   * Create relevant qpsLimit dynamicProperty and watch the configuration change.
+   * Search and return a valid qpsController.
+   */
+  protected QpsController create(String qualifiedNameKey) {
+    if (!QUALIFIED_KEY_CHECKER.matcher(qualifiedNameKey).matches()) {
+      throw new IllegalArgumentException("Unexpected qualified name: [" + qualifiedNameKey + "]");
+    }
+    // create "microservice"
+    createQpsControllerIfNotExist(qualifiedNameKey.substring(0, qualifiedNameKey.indexOf(SEPARATOR)));
+    // create "microservice.schema"
+    createQpsControllerIfNotExist(qualifiedNameKey.substring(0, qualifiedNameKey.lastIndexOf(SEPARATOR)));
```

Comment:
We need to add naming conventions to docs of service name, schema and operation

Reply:
OK. I will add naming convention instructions in gitbook later.